tests/e2e/tests/devfiles/DotNetCore.spec.ts

```diff
@@ -0,0 +1,78 @@
+ * Copyright (c) 2020 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+import 'reflect-metadata';
+import { WorkspaceNameHandler, Editor, CLASSES } from '../..';
+import { e2eContainer } from '../../inversify.config';
+import * as projectAndFileTests from '../../testsLibrary/ProjectAndFileTests';
+import * as commonLsTests from '../../testsLibrary/LsTests';
+import * as workspaceHandling from '../../testsLibrary/WorksapceHandlingTests';
+import * as codeExecutionTests from '../../testsLibrary/CodeExecutionTests';
+
+const editor: Editor = e2eContainer.get(CLASSES.Editor);
+
+const workspaceSampleName: string = 'dotnet-web-simple';
+const fileFolderPath: string = `${workspaceSampleName}`;
+const tabTitle: string = 'Program.cs';
+// const codeNavigationClassName: string = '[metadata] Console.cs';
+const stack : string = '.NET Core';
+const updateDependenciesTaskName: string = 'update dependencies';
+const buildTaskName: string = 'build';
+const runTaskName: string = 'run';
+const runTaskNameExpectedString: string = 'A process is now listening on port 5000.';
+
+suite(`${stack} test`, async () => {
+    suite (`Create ${stack} workspace`, async () => {
+        workspaceHandling.createAndOpenWorkspace(stack);
+        projectAndFileTests.waitWorkspaceReadinessNoSubfolder(workspaceSampleName);
+    });
+
+    suite('Test opening file', async () => {
+        // opening file that soon should give time for LS to initialize
+        projectAndFileTests.openFile(fileFolderPath, tabTitle);
+        prepareEditorForLSTests();
+    });
+
+    suite('Validation of workspace build and run', async () => {
+        codeExecutionTests.runTask(updateDependenciesTaskName, 120_000);
+        codeExecutionTests.closeTerminal(updateDependenciesTaskName);
+        codeExecutionTests.runTask(buildTaskName, 30_000);
+        codeExecutionTests.closeTerminal(buildTaskName);
+        codeExecutionTests.runTaskWithDialogShellAndOpenLink(runTaskName, runTaskNameExpectedString , 30_000);
+    });
+
+    suite('Language server validation', async () => {
+        commonLsTests.suggestionInvoking(tabTitle, 22, 33, 'test');
+        commonLsTests.errorHighlighting(tabTitle, 'error_text;', 23);
+        commonLsTests.autocomplete(tabTitle, 22, 27, 'WriteLine');
+        // commonLsTests.codeNavigation(tabTitle, 22, 27, codeNavigationClassName); // codenavigation is inconsistent https://github.com/eclipse/che/issues/16929
+    });
+
+    suite ('Stopping and deleting the workspace', async () => {
+        let workspaceName = 'not defined';
+        suiteSetup( async () => {
+            workspaceName = await WorkspaceNameHandler.getNameFromUrl();
+        });
+        test (`Stop worksapce`, async () => {
```

Comment:
Do we really need to check workspace stoping?

Reply:
that's a cleanup step that's done in every single devfile test. we need to remove running workspaces because we run against single deployment that only allows one workspace per user.

Reply:
So, in this case, maybe better to remove this step and use only "Delete workspace" step, it should work for running workspace.

Reply:
Anyway, this is only for your choice.

Reply:
I think I would leave it as is for now, we could consider it later and open a separate PR, because the change will be required in every devfile class.